grpc/service_config/service_config.proto

```diff
@@ -0,0 +1,315 @@
+  // addresses.  If unset, defaults to round_robin.
+  // Currently, the only supported values are round_robin and pick_first.
+  // Note that this will be used both in balancer mode and in fallback mode.
+  // Multiple LB policies can be specified; clients will iterate through
+  // the list in order and stop at the first policy that they support.
+  repeated LoadBalancingConfig child_policy = 1;
+}
+
+// Configuration for xds LB policy.
+message XdsConfig {
+  // Required.  Name of balancer to connect to.
+  string balancer_name = 1;
+  // Optional.  What LB policy to use for intra-locality routing.
+  // If unset, will use whatever algorithm is specified by the balancer.
+  // Multiple LB policies can be specified; clients will iterate through
+  // the list in order and stop at the first policy that they support.
+  repeated LoadBalancingConfig child_policy = 2;
+  // Optional.  What LB policy to use in fallback mode.  If not
+  // specified, defaults to round_robin.
+  // Multiple LB policies can be specified; clients will iterate through
+  // the list in order and stop at the first policy that they support.
+  repeated LoadBalancingConfig fallback_policy = 3;
+}
+
+// Selects LB policy and provides corresponding configuration.
+//
+// In general, all instances of this field should be repeated.
+// Clients will iterate through the list in order and stop at the first
+// policy that they support.  This allows the service config to specify
+// custom policies that may not be known to all clients.
+message LoadBalancingConfig {
+  // Exactly one LB policy may be configured.
+  oneof policy {
+    // For each new LB policy supported by gRPC, a new field must be added
+    // here.  The field's name must be the LB policy name and its type is a
+    // message that provides whatever configuration parameters are needed
+    // by the LB policy.  The configuration message will be passed to the
+    // LB policy when it is instantiated on the client.
+    //
+    // If the LB policy does not require any configuration parameters, the
+    // message for that LB policy may be empty.
+    //
+    // Note that if an LB policy contains another nested LB policy
+    // (e.g., a gslb policy picks the cluster and then delegates to
+    // a round_robin policy to pick the backend within that cluster), its
+    // configuration message may include a nested instance of the
+    // LoadBalancingConfig message to configure the nested LB policy.
+    PickFirstConfig pick_first = 4 [json_name = "pick_first"];
+    RoundRobinConfig round_robin = 1 [json_name = "round_robin"];
+    // xds field should not be used unless people know what they're
+    // doing.
+    XdsConfig xds = 2;
+    GrpcLbConfig grpclb = 3;
+    // xds_experimental field should only be used for testing the xds
+    // lbpolicy on the gRPC client.
+    // TODO(rekarthik): remove this field after xds migration is done.
+    XdsConfig xds_experimental = 5 [json_name = "xds_experimental"];
```

Comment:
Should this be removed?

Reply:
I think it's fine to leave it here.  Given the name "experimental", anyone who tries to use it will not be expecting anything to work.

Note that once this change gets imported, I intend to replace our internal copy of this proto with this one, so the two need to be the same.  We are using the xds_experimental field during development.